examples/training/classification/imagenet/basic_training.py

```diff
@@ -182,17 +202,72 @@ def augment(img, label):
         **eval(FLAGS.model_kwargs),
     )
 
-
 """
-Next, we pick an optimizer. Here we use Adam with a constant learning rate.
-Note that learning rate will decrease over time due to the ReduceLROnPlateau callback.
+Optional LR schedule with cosine decay instead of ReduceLROnPlateau
 """
+def lr_warmup_cosine_decay(global_step,
```

Comment:
Is this different from https://www.tensorflow.org/api_docs/python/tf/keras/optimizers/schedules/CosineDecayRestarts ?

Reply:
Yes - it doesn't restart. It goes globally from 0 to the initial rate, and then slowly lowers it from there (after an optional hold)

Reply:
![scheduler](https://user-images.githubusercontent.com/60978046/195218198-b975a3da-c6a4-4aca-9a90-da2640ae8d81.png)

Reply:
@ianstenbit mentioned swings in valid accuracy, which were resolved with a `clipnorm` - though, a warmup LR could also help IMO. I generally prefer using `ReduceLROnPlateau`, since most networks I trained seemed to do better with it than with a scheduler like this _but_ it might be worth testing

Reply:
I'm bringing warm up lr to Keras Core, so hopefully this will be temporary. Can you add a TODO here?

Reply:
fyi, a reference. of `WarmupLearningRateSchedule`, [script [line 34]](https://www.kaggle.com/code/ipythonx/tf-hybrid-efficientnet-swin-transformer-gradcam?scriptVersionId=107559761&cellId=54). It would be great to have it in core keras.

Reply:
@tanzhenyu Glad to hear you're bringing it to core Keras - it really felt like it was needed. I'll add a note that this is a temporary scheduler that'll be replaced later once the core Keras update goes out.

@innat Looks good! Do you want me to replace mine with this due to other decay schedules being supported as well?

Reply:
Added the TODO

Reply:
> Yes - it doesn't restart. It goes globally from 0 to the initial rate, and then slowly lowers it from there (after an optional hold)

I meant if there was some specific improvement in your experiment against that.

Reply:
Oh - I haven't benchmarked them against each other yet. I didn't have much luck with cyclical LR schedules before, and by extension, cosine with repeats. I think that this implementation is almost verbatim the same as the official one.
I really like @innat's implementation since it lets you switch between decay schedules instead of having only a fixed one. Thinking about maybe porting it into the script but adding the optional `hold` as well, since IMHO, it usually makes sense to hold at the initial LR at least for a bit